### DISCUSSION TITLE
Proposal for 'fs.embed' setting in Default Configuration for Gowebly

### DISCUSSION THREAD

**Discussion by abuffery** [gowebly/gowebly#40] [Ideas]
**1. Overview:**

The proposal involves extending the capabilities of the Gowebly default configuration file by adding settings to leverage 'fs.embed' for both templates and static files. This enhancement will enable users to seamlessly integrate embedded file systems into their projects, resulting in more efficient resource management and enhanced web application performance.

**2. Key Features:**

**a. Template Integration:**
   - Add a 'templateEmbed' setting in the configuration file, allowing users to specify templates that should be embedded using 'fs.embed.'
   - Support for multiple template files or directories to be embedded.

**b. Static File Integration:**
   - Introduce a 'staticEmbed' setting in the configuration file for embedding static files.
   - Specify which static files or directories should be embedded using 'fs.embed.'

**c. Flexibility and Customization:**
   - Allow users to choose whether to embed templates, static files, or both.

**3. User Benefits:**

   - **Improved Performance:** Embedded files reduce disk I/O, leading to faster load times.
   - **Simplified Deployment:** Eliminate the need to manage external template and static files separately.
   - **Reduced Configuration Complexity and Setup:** Simplify deployment pipelines and configuration files.

**4. Implementation Considerations:**

   - Ensure compatibility with existing Gowebly functionality and user workflows.
   - Handle edge cases gracefully, such as large-scale projects with extensive resources.
   - Provide backward compatibility for projects using older configurations.

**5. Rollout Plan:**

   - Develop and test the feature internally to ensure its stability and reliability.
   - Conduct a beta release to gather user feedback and identify any potential issues.
   - Gradually introduce the enhanced default configuration options to all Gowebly users through documentation and updates.

**6. Conclusion:**

The addition of 'fs.embed' integration to the default configuration file for Gowebly can represent a significant enhancement for  users. This feature aligns Gowebly with modern web development options, offering users the power to optimize their projects for better performance and organization when embedding is the best option. This addition would help Gowebly to position itself as a versatile and user-friendly web development tool.

**Comment by koddr** [comment]
Hi, @abuffery 

Since `v2.0.0`, the Gowebly CLI has moved to TUI, it no longer has the configuration file that it had before.

I've already included statics in the binary, but that only works now for the backend on the built-in net/http package, Chi and HttpRouter. If you have time, you can help me implement the same for Fiber, Gin, and Echo web frameworks. I would be very happy to do so!

Embedding HTML templates is a bit more complicated, as I mostly use Templ in my projects and am not very familiar with embedding regular templates in a binary.
